lib/cylc/config.py

```diff
@@ -1688,17 +1688,10 @@ def get_graph_raw(self, start_point_string, stop_point_string,
             stop_point = None
 
         # For nested families, only consider the outermost one
-        clf = copy(self.closed_families)
-        for i in self.closed_families:
-            for j in self.closed_families:
-                if i in members[j]:
-                    if i in clf:
-                        clf.remove(i)
-
         clf_map = {}
-        for family_name in clf:
-            clf_map[family_name] = (
-                self.runtime['first-parent descendants'][family_name])
+        for name in self.closed_families:
+            if any([name not in members[i] for i in self.closed_families]):
```

Comment:
This if statement only evaluates once the entire list has been constructed, switch to a generator to accelerate this https://github.com/metomi/rose/pull/2064#discussion_r112664279.